firmware/shared/code/messageHandler.c

```diff
@@ -26,74 +31,152 @@ typedef struct
+    for (messageHandler_RXMessageChannel_E channel = (messageHandler_RXMessageChannel_E)0U; channel < MESSAGE_HANDLER_RX_CHANNEL_COUNT; channel++)
     {
+        const messageHandler_RXMessageConfig_S * const RXChannelConfig = &messageHandler_config.RXMessageConfig[channel];
+        messageHandler_RXChannelData_S * const RXChannelData = &messageHandler_data.RXChannelData[channel];
+        
         // Init message
-        memcpy(&messageHandler_data.RXChannelData[channel].message, &messageHandler_config.RXMessageConfig[channel].initValue, sizeof(messageHandler_config.RXMessageConfig[channel].initValue));
+        memcpy(&RXChannelData->message, &RXChannelConfig->initValue, sizeof(RXChannelConfig->initValue));
+#if USE_CAN
+        // Add RX message to filter, use RX message channel as filter number
+        CAN_filterAdd(RXChannelConfig->messageID, (uint16_t)channel);
+#endif
+    }
+#if USE_CAN
+    for (messageHandler_TXMessageChannel_E channel = (messageHandler_TXMessageChannel_E)0U; channel < MESSAGE_HANDLER_TX_CHANNEL_COUNT; channel++)
+    {
+        messageHandler_TXChannelData_S * const TXChannelData = &messageHandler_data.TXChannelData[channel];
+
+        TXChannelData->dispatchRequested = true;
     }
+#endif
 }
 
 void messageHandler_run1ms(void)
 {
     // RX
-    for(messageHandler_RXMessageChannel_E channel = (messageHandler_RXMessageChannel_E)0U; channel < MESSAGE_HANDLER_RX_MESSAGE_CHANNEL_COUNT; channel++)
+    for(messageHandler_RXMessageChannel_E channel = (messageHandler_RXMessageChannel_E)0U; channel < MESSAGE_HANDLER_RX_CHANNEL_COUNT; channel++)
     {
         messageHandler_RXChannelData_S * const channelData = &messageHandler_data.RXChannelData[channel];
+        const messageHandler_RXMessageConfig_S * const channelConfig = &messageHandler_config.RXMessageConfig[channel];
 
         if(channelData->newDataAvailable)
         {
-            if(messageHandler_config.messageReceivedCallback != NULL)
+            if(channelConfig->callbackEnable)
             {
-                messageHandler_config.messageReceivedCallback(channel, &channelData->message);
+                if(messageHandler_config.messageReceivedCallback != NULL)
+                {
+                    messageHandler_config.messageReceivedCallback(channel, &channelData->message);
+                }
             }
 
             channelData->newDataAvailable = false;
         }
     }
 
     // TX
-    for(messageHandler_TXMessageChannel_E channel = (messageHandler_TXMessageChannel_E)0U; channel < MESSAGE_HANDLER_TX_MESSAGE_CHANNEL_COUNT; channel++)
+    for(messageHandler_TXMessageChannel_E channel = (messageHandler_TXMessageChannel_E)0U; channel < MESSAGE_HANDLER_TX_CHANNEL_COUNT; channel++)
     {
         messageHandler_TXChannelData_S * const channelData = &messageHandler_data.TXChannelData[channel];
         const messageHandler_TXMessageConfig_S * const channelConfig = &messageHandler_config.TXMessageConfig[channel];
+
+#if USE_CAN
```

Comment:
at the top of the file, maybe include how to enable pre-processor commands.

Reply:
I'm just going to get rid of this and not have support for UART. We'll never use it. It's just adding unnecessary complexity.